starters/web-service/templates/ingress.yaml

```diff
@@ -0,0 +1,29 @@
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: "<CHARTNAME>"
+  labels:
+    {{- include "<CHARTNAME>.labels" . | nindent 4 }}
+  annotations:
+    kubernetes.io/ingress.class: "nginx"
+    {{- if .Values.ingress.gafaelfawrAuthQuery }}
+    nginx.ingress.kubernetes.io/auth-method: "GET"
+    nginx.ingress.kubernetes.io/auth-response-headers: "X-Auth-Request-User"
+    nginx.ingress.kubernetes.io/auth-signin: "{{ .Values.global.baseUrl }}/login"
+    nginx.ingress.kubernetes.io/auth-url: "{{ .Values.global.baseUrl }}/auth?{{ .Values.ingress.gafaelfawrAuthQuery }}"
+    {{- end }}
+    {{- with .Values.ingress.annotations }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+spec:
+  rules:
+    - host: {{ required "global.host must be set" .Values.global.host | quote }}
+      http:
+        paths:
+          - path: "/<CHARTNAME>"
```

Comment:
Something I've found very helpful is to do:

```yaml
- path: {{ .Values.ingress.path }}
```

That way I can also include ``Values.ingress.path`` in my app's config map to mount it on a consistent path.

For example, Times Square is at `/times-square/api`, which isn't its chart name, so that I can let `/times-square` be a URL served by Squareone.